### PR TITLE
Abstract

### PR DESCRIPTION
**Pull request by Ambrevar**

### CODE REVIEW

**Comment by Ambrevar** [comment]
@DamienCassou, let me know if this works for you with `dnf`.
This is a first draft, there will be more changes to come.

File: helm-system-packages-guix.el [hunk 1]
```diff
@@ -302,31 +301,15 @@ Otherwise display in `helm-system-packages-buffer'."
...
-  (unless (helm-system-packages-missing-dependencies-p "guix" "recsel")
-    (helm :sources (helm-system-packages-guix-build-source)
-          :buffer "*helm guix*"
-          :truncate-lines t
-          :input (when helm-system-packages-use-symbol-at-point-p
-                   (substring-no-properties (or (thing-at-point 'symbol) ""))))))
+(defvar helm-system-packages-guix
+  (helm-system-packages-manager-create
+   :name "guix"
+   :refresh-function 'helm-system-packages-guix-refresh
```
Comment (DamienCassou) [inline helm-system-packages-guix.el:307]: you may use `#'` when quoting function names to tell the compiler it's a function

**Review by DamienCassou — APPROVED** [review]
it's exactly the kind of change I was expecting. Good job!

**Comment by DamienCassou** [comment]
I've just updated my PR to use the new abstraction. I like it very much. Please keep going.

**Comment by Ambrevar** [comment]
Merged.  Let me know if there is anything else you'd like to see factored.

**Pull request closed without merging**